pkg/cmd/backtest.go

```diff
@@ -324,28 +325,35 @@ var BacktestCmd = &cobra.Command{
 			klineChans = append(klineChans, KChanEx{KChan: c, Exchange: exchange})
 		}
 
-		for {
-			count := len(klineChans)
-			for _, kchanex := range klineChans {
-				kLine, more := <-kchanex.KChan
-				if more {
-					kchanex.Exchange.ConsumeKLine(kLine)
-				} else {
-					if err := kchanex.Exchange.CloseMarketData(); err != nil {
-						return err
+		shutdownCtx, cancelFunc := context.WithCancel(ctx)
```

Comment:
use a separate ctx name for this? how about:

```
runCtx, cancelRun := 
```

Reply:
done